pkg/sessionmgr/sessionmgr-design.md

```diff
@@ -0,0 +1,389 @@
+
+#### sandboxMgr
+
+* Expose an HTTP API to `sessionMgr` to create sandboxes.
+* Responsibilities:
+
+    * Create the sandbox (via `RuntimeController`, etc.).
+    * Write the `sessionID` → `sandbox` mapping to Redis using:
+
+        * `SetSessionLockIfAbsent`
+        * `BindSessionWithSandbox`
+        * `DeleteSessionBySandboxIDTx`
+    * Return both `sessionID` and sandbox data back to `sessionMgr`.
+
+#### Redis
+
+* Store the `sessionID -> Sandbox` mapping with TTL.
+* Provide a Redis client. In this design, `sessionMgr` uses Redis only through `GetSandboxBySessionID`.
+
+### 1.3 Sequence Diagram
+
+![Session manager sequence diagram](./sessionmgr-sequence.png)
+
+---
+
+## 2. API Design
+
+### 2.1 Router ➜ sessionMgr
+
+```go
+// Interface exposed to the Router.
+type Manager interface {
+    GetSandboxBySession(ctx context.Context, req *GetSandboxBySessionRequest) (*GetSandboxBySessionResponse, error)
+}
+
+// Request from Router.
+type GetSandboxBySessionRequest struct {
+    // SessionID passed from upstream; may be empty.
+    SessionID string
+
+    // Metadata used when a new sandbox needs to be created.
+    Kind      string // agent type, e.g. "code-interpreter"
+    Name      string // logical agent or business name, used for logging and sandbox naming
+    Namespace string // k8s namespace / business namespace
+
+    // Optional extension fields: tenant/user/resource/model/etc.
+    // TenantID string
+    // ...
+}
+
+// Response to Router.
+type GetSandboxBySessionResponse struct {
+    SessionID string // always non-empty; if the original SessionID was empty, this is the newly created one
+    Endpoint  string // sandbox endpoint (e.g. "http://ip:port" or a gRPC address)
+}
+```
+
+**Notes:**
+
+* When `SessionID` is **empty**, `response.SessionID` is a **new** session ID; the Router must carry it in subsequent requests.
+* When `SessionID` is **non-empty**, if the session exists, `sessionMgr` returns the **same** `SessionID` and its corresponding `Endpoint`.
+
+### 2.2 sessionMgr ➜ Redis
+
+`sessionMgr` uses a **read-only** subset of the Redis client:
+
+```go
+type RedisClient interface {
+    GetSandboxBySessionID(ctx context.Context, sessionID string) (*Sandbox, error)
+}
+
+// Sandbox structure stored in Redis (recommended to be defined in a shared module).
+type Sandbox struct {
+    ID       string // sandboxID
+    Endpoint string // actual endpoint to send traffic to
+    // additional metadata...
+}
+```
+
+**Error contract (recommended):**
+
+* `GetSandboxBySessionID` finds a record:
+  `(*Sandbox, nil)`
+* Not found:
+  `(nil, ErrSessionNotFound)` — this should be a wrapped, domain-level error, not a raw Redis driver error.
+* Other Redis errors:
+  `(nil, error)`.
+
+### 2.3 sessionMgr ➜ sandboxMgr (HTTP API)
+
+We recommend a simple REST API for creating sandboxes.
+
+* **URL**: `POST /v1/sandboxes`
+* **Request body (JSON)**:
+
+```jsonc
+{
+  "kind": "code-interpreter",
```

Comment:
I would suggest, we use the kind directly from api. should be `CodeInterpreter` 
FYI https://github.com/volcano-sh/agentcube/pull/45/files#diff-8d4fdf9be50480fe81a50076275bb5e5f07a5a76bfeae89544962c5dcab8f682R9